src/storage/src/source/reclock.rs

```diff
@@ -130,7 +130,8 @@ where
 
         while *self.upper == [IntoTime::minimum()]
             || (PartialOrder::less_equal(&self.source_upper.frontier(), &new_from_upper)
-                && PartialOrder::less_than(&self.upper, &new_into_upper))
+                && PartialOrder::less_than(&self.upper, &new_into_upper)
+                && self.upper.less_equal(&binding_ts))
```

Comment:
Just flagging that it's not clear to me why this was changed, though it might be clear to someone with more context on the reclock operator.

Reply:
This is a bug fix. We want the mint operation to be a no-op if either `new_from_upper` or `binding_ts` are not beyond the current `source_upper` and `upper` respectively. The code on main only checked the former which meant that if the requested `binding_ts` was in the past we would attempt to write to the remap shard updates that are not beyond the upper.